gravitee-node-api/src/main/java/io/gravitee/node/api/server/Server.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.node.api.server;
+
+import io.gravitee.node.api.certificate.KeyStoreManager;
+import io.gravitee.node.api.certificate.TrustStoreManager;
+
+/**
+ * @author Jeoffrey HAEYAERT (jeoffrey.haeyaert at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public interface Server<C extends ServerOptions> {
```

Comment:
Should we anticipate the need for a "type" such as HTTP, TCP etc ?

Reply:
Indeed. I've added a `String type()`  method on the Server interface 👍